### DISCUSSION TITLE
Articulation link index

### DISCUSSION THREAD

**Discussion by zemlifr** [NVIDIA-Omniverse/PhysX#255] [Q&A] [answered]
`PxArticulationLink::getLinkIndex` says that it returns 0xFFFFFFFF when articulation is not in scene. What is a reason for this behavior?
I am using getLinkIndex to manipulate articulation cache when articulation is outside of scene. This works fine in physx 4.1, but now I am trying to update to physx 5.3 and it does not work anymore.

Are indices recalculated when articulation is added to scene? If so how should I handle such situations?
I am talking about situation when articulation previously was in scene and then it was removed so indices were set previously.

**Comment by preist-nvidia** [comment, marked as answer]
Hi @zemlifr - yes exactly, we calculate the index when the articulation is added to the scene, and the index is not necessarily identical to the order of the link being added to the articulation.

Are you changing the articulation topology when removing and readding to the scene? If not, the indices should be the same.

**Reply by zemlifr** [reply]
Thanks for quick answer. I am not changing articulation topology, I only change joint positions when articulation is not in scene.
So from your answer I will assume that I will be safe if I store LinkIndex once articulation is added to scene first time.
